src/Bybit/Bybit.jl

```diff
@@ -0,0 +1,180 @@
+
+Client info.
+
+## Required fields
+- `base_url::String`: Base URL for the client. 
+
+## Optional fields
+- `public_key::String`: Public key for authentication.
+- `secret_key::String`: Secret key for authentication.
+- `interface::String`: Interface for the client.
+- `proxy::String`: Proxy information for the client.
+- `account_name::String`: Account name associated with the client.
+- `description::String`: Description of the client.
+"""
+Base.@kwdef struct BybitClient <: AbstractAPIsClient
+    base_url::String
+    public_key::Maybe{String} = nothing
+    secret_key::Maybe{String} = nothing
+    interface::Maybe{String} = nothing
+    proxy::Maybe{String} = nothing
+    account_name::Maybe{String} = nothing
+    description::Maybe{String} = nothing
+end
+
+"""
+    BybitAPIError{T} <: AbstractAPIsError
+
+Exception thrown when an API method fails with code `T`.
+
+## Required fields
+- `retCode::Int64`: Error code.
+- `retMsg::String`: Error message.
+- `time::NanoDate`: Error time.
+
+## Optional fields
+- `result::String`: Error result.
+- `retExtInfo::String`: Extended error information.
+"""
+struct BybitAPIError{T} <: AbstractAPIsError
+    retCode::Int64
+    retMsg::String
+    result::Dict{String,Any}
+    retExtInfo::Dict{String,Any}
+    time::NanoDate
+
+    function BybitAPIError(retCode::Int64, x...)
+        @assert !iszero(retCode) # retCode = 0 means success
+        return new{retCode}(retCode, x...)
+    end
+end
+
+CryptoAPIs.error_type(::BybitClient) = BybitAPIError
+
+function Base.show(io::IO, e::BybitAPIError)
+    return print(io, "code = ", "\"", e.retCode, "\"", ", ", "msg = ", "\"", e.retMsg, "\"")
+end
+
+function CryptoAPIs.request_sign!(::BybitClient, query::Q, ::String)::Q where {Q<:BybitPublicQuery}
+    return query
+end
+
+function time2unix(x::DateTime; digits::Real = 1000)
+    return round(Int64, digits * datetime2unix(x))
+end
+
+function CryptoAPIs.request_sign!(client::BybitClient, query::Q, ::String)::Q where {Q<:BybitPrivateQuery}
+    query.timestamp = Dates.now(UTC)
+    query.api_key = client.public_key
+    query.sign = nothing
+    body::String = Serde.to_query(query)
+    salt = join([string(time2unix(query.timestamp)), client.public_key, query.recv_window, body])
+    query.sign = hexdigest("sha256", client.secret_key, salt)
+    return query
+end
+
+function CryptoAPIs.request_body(::Q)::String where {Q<:BybitCommonQuery}
+    return ""
+end
+
+function CryptoAPIs.request_query(query::Q)::String where {Q<:BybitCommonQuery}
+    return Serde.to_query(query)
+end
+
+function CryptoAPIs.request_headers(client::BybitClient, ::BybitPublicQuery)::Vector{Pair{String,String}}
+    return Pair{String,String}[
+        "Content-Type" => "application/json"
+    ]
+end
+
+function CryptoAPIs.request_headers(client::BybitClient, query::BybitPrivateQuery)::Vector{Pair{String,String}}
+    return Pair{String,String}[
+        "X-BAPI-SIGN-TYPE" => "2",
+        "X-BAPI-SIGN" => query.sign,
```

Comment:
in all private methods, replace `sign` with `signature`, to maintain uniformity between exchanges